shared/src/business/useCases/docketEntry/fileCourtIssuedDocketEntryInteractor.js

```diff
@@ -31,128 +31,160 @@ exports.fileCourtIssuedDocketEntryInteractor = async (
       applicationContext,
-      docketNumber,
+      docketNumber: subjectDocketNumber,
     });
 
-  let caseEntity = new Case(caseToUpdate, { applicationContext });
+  let subjectCaseToUpdateEntity = new Case(subjectCaseToUpdate, {
+    applicationContext,
+  });
 
-  const docketEntry = caseEntity.getDocketEntryById({
+  const subjectDocketEntry = subjectCaseToUpdateEntity.getDocketEntryById({
     docketEntryId,
   });
 
-  if (!docketEntry) {
+  if (!subjectDocketEntry) {
     throw new NotFoundError('Docket entry not found');
   }
-  if (docketEntry.isOnDocketRecord) {
+
+  if (subjectDocketEntry.isOnDocketRecord) {
     throw new Error('Docket entry has already been added to docket record');
   }
 
-  const user = await applicationContext
-    .getPersistenceGateway()
-    .getUserById({ applicationContext, userId: authorizedUser.userId });
-
   const numberOfPages = await applicationContext
     .getUseCaseHelpers()
     .countPagesInDocument({ applicationContext, docketEntryId });
 
-  const isUnservable = UNSERVABLE_EVENT_CODES.includes(documentMeta.eventCode);
+  const user = await applicationContext
+    .getPersistenceGateway()
+    .getUserById({ applicationContext, userId: authorizedUser.userId });
 
-  const docketEntryEntity = new DocketEntry(
-    {
-      ...omit(docketEntry, 'filedBy'),
-      attachments: documentMeta.attachments,
-      date: documentMeta.date,
-      documentTitle: documentMeta.generatedDocumentTitle,
-      documentType: documentMeta.documentType,
-      draftOrderState: null,
-      editState: JSON.stringify(documentMeta),
-      eventCode: documentMeta.eventCode,
-      filingDate: documentMeta.filingDate,
-      freeText: documentMeta.freeText,
-      isDraft: false,
-      isFileAttached: true,
-      isOnDocketRecord: true,
-      judge: documentMeta.judge,
-      numberOfPages,
-      scenario: documentMeta.scenario,
-      serviceStamp: documentMeta.serviceStamp,
-      trialLocation: documentMeta.trialLocation,
-      userId: user.userId,
-    },
-    { applicationContext },
-  );
-
-  const workItem = new WorkItem(
-    {
-      assigneeId: null,
-      assigneeName: null,
-      associatedJudge: caseToUpdate.associatedJudge,
-      caseStatus: caseToUpdate.status,
-      caseTitle: Case.getCaseTitle(caseEntity.caseCaption),
-      docketEntry: {
-        ...docketEntryEntity.toRawObject(),
-        createdAt: docketEntryEntity.createdAt,
-      },
-      docketNumber: caseToUpdate.docketNumber,
-      docketNumberWithSuffix: caseToUpdate.docketNumberWithSuffix,
-      hideFromPendingMessages: true,
-      inProgress: true,
-      section: DOCKET_SECTION,
-      sentBy: user.name,
-      sentByUserId: user.userId,
-    },
-    { applicationContext },
-  );
+  const isUnservable = UNSERVABLE_EVENT_CODES.includes(documentMeta.eventCode);
 
-  if (isUnservable) {
-    workItem.setAsCompleted({ message: 'completed', user });
-  }
+  for (let docketNumber of docketNumbers) {
```

Comment:
For each case, we run the multiple saves asynchronously, but we iterate over each case synchronously.  Especially for large consolidated case groups, we should asynchronously iterate over each case.  That's what we did in `fileAndServeCourtIssuedDocumentInteractor`.

Reply:
Was any thought put into moving the similar code between this interactor and the `fileAndServeCourtIssuedDocumentInteractor` we touched into a usecase helper?

Reply:
I haven't checked if there were ways to re-use code.  I can try double checking.

Reply:
I checked the code and the code paths seem to do pretty different things.  I feel drying these up would result in more complexity